src/coprocessor/statistics/analyze.rs

```diff
@@ -130,9 +146,37 @@ impl<S: Snapshot> AnalyzeContext<S> {
                     cms.insert(&data);
                 }
             }
+            if stats_version == ANALYZE_VERSION_V2 {
+                if cur_val.1 == data {
+                    cur_val.0 += 1;
+                } else {
+                    if cur_val.0 > 0 {
+                        topn_heap.push(Reverse(cur_val));
+                    }
+                    if topn_heap.len() > top_n_size {
+                        topn_heap.pop();
+                    }
+                    cur_val = (1, data.clone());
+                }
+            }
             hist.append(&data);
```

Comment:
Funny `clone` hunting. XD

```suggestion
            hist.append(&data);
            if stats_version == ANALYZE_VERSION_V2 {
                if cur_val.1 == data {
                    cur_val.0 += 1;
                } else {
                    if cur_val.0 > 0 {
                        topn_heap.push(Reverse(cur_val));
                    }
                    if topn_heap.len() > top_n_size {
                        topn_heap.pop();
                    }
                    cur_val = (1, data);
                }
            }
```

Reply:
@winoros  🤣